identity-iota/src/credential/errors.rs

```diff
@@ -0,0 +1,127 @@
+                                                                 * here? */
+  },
+
+  /// Indicates that the presentation's holder could not be parsed as a valid DID.
+  #[error("the presentation's holder property could not be parsed to a valid DID")]
+  #[non_exhaustive]
+  HolderUrl {
+    source: Box<dyn std::error::Error + Send + Sync + 'static>, /* Todo: Would it be better to use a specific type
+                                                                 * here? */
+  },
+
+  /// Indicates an attempt to verify a presentation's signature with a resolved DID document not corresponding to the
+  /// URL of the presentation's holder property.
+  #[error("the provided holder document does not correspond to the presentation's holder property")]
+  #[non_exhaustive]
+  IncompatibleHolderDocument,
+
+  /// Indicates that the presentation's signature could not be verified using the holder's DID Document.
+  #[error("could not verify the holder's signature")]
+  #[non_exhaustive]
+  HolderProof {
+    source: Box<dyn std::error::Error + Send + Sync + 'static>, /* Todo: Would it be better to use a specific type
+                                                                 * here? */
+  },
+  /// Indicates that the structure of the [Credential](identity_credential::credential::Credential) is not semantically
+  /// correct.
+  #[error("the credential's structure is not semantically correct")]
+  CredentialStructure(#[source] identity_credential::Error),
+  /// Indicates that the structure of the [Presentation](identity_credential::presentation::Presentation) is not
+  /// semantically correct.
+  #[error("the presentation's structure is not semantically correct")]
+  PresentationStructure(#[source] identity_credential::Error),
+  /// Indicates that the relationship between the presentation holder and one of the credential subjects is not valid.
+  #[error("expected holder = subject of the credential at position {credential_position}")]
+  #[non_exhaustive]
+  InvalidHolderSubjectRelationship { credential_position: usize },
+  /// Indicates that the presentation does not have a holder.
+  #[error("the presentation has an empty holder property")]
+  MissingPresentationHolder,
+}
+
+// Todo: Consider implementing Error for OneOrMany<E: std::error::Error> to avoid wrapping it in
+// AccumulatedCredentialValidationError
+#[derive(Debug)]
+/// An error caused by a failure to validate a Credential.  
+pub struct AccumulatedCredentialValidationError {
+  pub validation_errors: OneOrMany<ValidationError>,
+}
+
+impl Display for AccumulatedCredentialValidationError {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    // intersperse might become available in the standard library soon: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.intersperse
+    let detailed_information: String = itertools::intersperse(
+      self.validation_errors.iter().map(|err| err.to_string()),
+      ", ".to_string(),
+    )
```

Comment:
Comma isn't a great delimiter here in my opinion, perhaps a semi-colon and/or new-line too?

How does this look when accumulated and reported with `anyhow`?

Reply:
Changed the display implementation now. 

Here is an example of what it looks like with `anyhow`:  
The following code 
```
let error = validator
      .validate(&credential, &options, &other_issuer_resolved_doc, fail_fast)
      .unwrap_err(); 

    let report = anyhow::Error::from(error);
    println!("anyhow report without causes:");
    println!("{}", report);
    println!("");
    println!("anyhow report with causes:");
    println!("{:#}", report);
```
produces the following output 

```
anyhow report without causes:
credential validation failed

anyhow report with causes:
credential validation failed: [issuance date is in the future or later than required; the expiration date is in the past or earlier than required; the credential's structure is not semantically correct; issuer did not match any of the supplied DID Documents]
```

Reply:
> Ideally a multi-line error report might be easier to read but I'm not sure how nicely that plays with loggers and the Wasm bindings.

In my experience most loggers prefer a one-line error report.